Matchismo/ViewController.m

```diff
@@ -6,17 +6,44 @@
 //
 
 #import "ViewController.h"
+#import "PlayingCardDeck.h"
 
 @interface ViewController ()
-
+@property (weak, nonatomic) IBOutlet UILabel *flipsLabel;
+@property (nonatomic) NSUInteger flipCount;
+@property (nonatomic) PlayingCardDeck *playingCardsDeck;
 @end
 
 @implementation ViewController
 
-- (void)viewDidLoad {
-    [super viewDidLoad];
-    // Do any additional setup after loading the view.
+- (PlayingCardDeck *)playingCardsDeck
+{
+    if (!_playingCardsDeck) _playingCardsDeck =  [PlayingCardDeck new];
+    return _playingCardsDeck;
+}
+
+- (void)setFlipCount:(NSUInteger)flipCount
+{
+    _flipCount = flipCount;
+    self.flipsLabel.text = [NSString stringWithFormat: @"Flips: %lu", self.flipCount];
+    NSLog(@"flipCount = %lu", self.flipCount);
 }
 
+- (IBAction)touchCardButton:(UIButton *)sender
+{
+    NSString *label = [self.playingCardsDeck drawRandomCard].contents;
+    if ([sender.currentTitle length] || ![label length]) {
```

Comment:
I am not sure checking based on the lengths of card title is the correct option for detecting which way the card is showing. 

- I have used a bool variable to check which way card shows
- Maybe you can use other option

Reply:
Can you elaborate? What exactly you think can go wrong? Cardback has empty string as a label and cardfront always has two symbols. And they use the same approach in the lecture

Reply:
The approach is totally okay. I don't see any issue. Seems the other approach might be slightly more appropriate.